pallets/claims/src/lib.rs

```diff
@@ -134,9 +149,17 @@ pub mod pallet {
 			let sender = ensure_signed(origin)?;
 
 			let (balance_due, address) = Self::validate_claim(&sender, &ethereum_signature)?;
-
-			Self::process_claim(sender, balance_due, address)?;
-
+			let vesting_check = Vesting::<T>::contains_key(&address);
+
+			if vesting_check {
```

Comment:
AS mentioned in previous comment, if you use OptionQuery, you can do something like :

```
If Some(vesting) = Vesting::<T>::get(&adress) {
...
}else{
...
}
```